lib/ui/geometry.dart

```diff
@@ -1438,38 +1451,38 @@ class RRect {
     if (point.dx < left || point.dx >= right || point.dy < top || point.dy >= bottom)
       return false; // outside bounding box
 
-    _scaleRadii();
+    final RRect scaled = _scaleRadii();
```

Comment:
since we're not caching that work anymore, this may have perf implications.
we should make sure we have a benchmark that covers the case here (of calling `contains` many times on the same rrects).

Reply:
https://github.com/flutter/flutter/pull/30985 will track that